okhttp/src/main/java/okhttp3/OkHttpClient.java

```diff
@@ -159,7 +159,16 @@ private OkHttpClient(Builder builder) {
     this.cache = builder.cache;
     this.internalCache = builder.internalCache;
     this.socketFactory = builder.socketFactory;
-    if (builder.sslSocketFactory != null) {
+
+    boolean isTLS = true;
+    for (ConnectionSpec spec : connectionSpecs) {
+      if (connectionSpecs.equals(DEFAULT_CONNECTION_SPECS)) {
+        break;
+      }
+      isTLS = spec.isTls();
+    }
+
+    if (builder.sslSocketFactory != null || !isTLS) {
```

Comment:
It's slightly unfortunate that you always need to loop even when a factory was specified. Trading one inefficiency for another (small) one.